### DISCUSSION TITLE
Data Quality Reports

### DISCUSSION THREAD

**Discussion by emi420** [hotosm/tasking-manager#5685] [Ideas]
Hi everyone! as part of the HOT efforts to improve the quality of the mapping, we're working on new tools for validation.

Previously, we discussed some ideas for validation [before uploading changes to the map](https://github.com/hotosm/tasking-manager/discussions/5617). 

This time, we'll discuss ideas for after uploading changes to the map.

## New feature: data quality reports

The proposal is to have a new section for each project, similar to stats, but for data quality, displaying maps and tables, and the option to download the data for loading into JOSM or to create challenges on Map Roulette.

We had been creating experimental data quality reports and shared them into spreadsheets on Google Drive. Now, we want to make this reports available into the Tasking Manager, for each project.

### Un-squared and overlapping buildings

The plan is to start with buildings and then add more priority features like roads and other data quality aspects like the semantic one (tagging).

<img width="997" alt="Screenshot 2023-04-10 at 18 29 08" src="https://user-images.githubusercontent.com/1226194/231002635-ad7a16a8-e7ae-4e1d-b9bb-c2f9c93aec34.png">

<img width="998" alt="Screenshot 2023-04-10 at 18 32 00" src="https://user-images.githubusercontent.com/1226194/231003062-f8817b52-cb3a-4950-89d1-c3df9c89c44e.png">

## Let's start discussing it and make the map a better map!

- What do you think?
- What are your concerns about a feature like this?
- Do you have any ideas to improve it?
- Do you think it's a bad idea or that it can be implemented different?
- Could be good to show this feature to all users? or just to validators and project owners?

Thanks :)

**Comment by adiatmad** [comment]
1. What do you think?
_Always good to share the ideas with the community. I like the way you communicate it_

2. What are your concerns about a feature like this?
_The amount of issues should be more meaningful and should prompt appropriate action._ 

3. Do you have any ideas to improve it?
_Probably an automatic notification for project managers if a specific amount of issues are reached. As a result, they can anticipate it earlier during the mapping process._

4. Do you think it's a bad idea or that it can be implemented different?
_Starting with the "unsquared building" issue and on to another prevalent issue is a great plan._

5. Could be good to show this feature to all users? or just to validators and project owners?
_I'd like to show this feature to all users so that everyone is on the same page regarding the data quality issue. Hopefully, they will be able to report any false positives, etc. in a timely manner._

**Reply by martien-176** [reply]
"Probably an automatic notification for project managers if a specific amount of issues are reached. As a result, they can anticipate it earlier during the mapping process."

-->> agree. Possibly also for validators?

**Comment by SColchester** [comment]
This is great @emi420 - really happy you are looking into this. At the Tasking Manager Collective meeting yesterday we discussed this and I want to get some of what we discussed down here. 

There is a lot of value on this information being available for all projects. But a key part to think carefully about is the desired next step for fixing the identified issues. 

**Raw GeoJSON download**

The option to download a GeoJSON of all bad objects is powerful but potentially problematic. I foresee a scenario with lazy project managers (including myself) downloading a GeoJSON of all un-square buildings and using JOSM’s squaring tool to 'fix' them all at once. (Or similar bulk ‘fixing’ methods for other features).  

This would likely cause more problems than it solves. Often un-squared buildings are poorly traced and need careful geometry fixes and sometimes buildings really don’t have squared corners.

The other issue with allowing a wide area GeoJSON download is that it risks causing edit conflicts.

**Show a coloured ‘validation priority’ task grid**

If the number of bad objects could be counted per task and then the tasks coloured on a scale relative to how many bad objects there are – this could allow validators to prioritize their validation efforts to tasks that have the most issues. Also by taking contributors back into the task structure it ensures that conflicts in the clean-up process are avoided. 

**MapRoulette Project**

Have the option to download a GeoJSON of points where each bad object is. This GeoJSON can then be used to generate a MapRoulette ‘clean-up’ project, where each small task in MapRoulette is to fix one building or to fix one bad highway intersection. 

To avoid editing conflicts, this GeoJSON download should only be possible once all activity has ceased on the project i.e. either the project is 100% mapped and validated, or it has been ‘Archived’.

**Reply by emi420** [reply]
I like the idea for MapRoulette. Actually, we did an experiment like that, we exported a bulk of possible un-squared buildings to create a challenge. But I didn't realize that just the points are needed, not the polygon. 

There will be a brainstorming for doing validating tasks on MapSwipe, this could fit also there.

Thanks Sam!

**Reply by martien-176** [reply]
"Show a coloured ‘validation priority’ task grid

If the number of bad objects could be counted per task and then the tasks coloured on a scale relative to how many bad objects there are – this could allow validators to prioritize their validation efforts to tasks that have the most issues. Also by taking contributors back into the task structure it ensures that conflicts in the clean-up process are avoided."

-->>  Such a "heatmapped" projectmap would be awful. If that is too hard to develop, a sortable table (one with bad data per task, one with bad data per mapper) could be an alternative.

**Comment by martien-176** [comment]
Would it be possible to integrate these data quality reports with eg HOT fAIr, MapSwipe or OSMCha?

**Reply by emi420** [reply]
Yes @martien-176 ! this is a separate project and could be integrated with other products.
